### PR TITLE
Separate storage, studio dependencies

### LINKED ISSUE #284: Incopatibility with sqlmodel and fastapi
Hi, loved ell, but I cant use it on my project unfortunately. 
Here are my dependencies
requires-python = ">=3.10"
dependencies = [
    "asyncpg>=0.29.0",
    "black>=24.8.0",
    "fastapi[standard]>=0.115.0",
    "numpy>=2.1.1",
    "pandas>=2.2.3",
    "psycopg2-binary>=2.9.9",
    "pydantic-settings>=2.5.2",
    "python-dotenv>=1.0.1",
    "scrapy>=2.11.2",
    "sqlmodel>=0.0.22",
]

when i tried to install ell-ai got this:

 No solution found when resolving dependencies for split (python_full_version == '3.10.*'):
  ╰─▶ Because only the following versions of ell-ai are available:
          ell-ai==0.0.1
          ell-ai==0.0.2
          ell-ai==0.0.3
          ell-ai==0.0.4
          ell-ai==0.0.5
          ell-ai==0.0.6
          ell-ai==0.0.7
          ell-ai==0.0.8
          ell-ai==0.0.9
          ell-ai==0.0.10
          ell-ai==0.0.11
          ell-ai==0.0.12
          ell-ai==0.0.13
      and all of:
          ell-ai<=0.0.7
          ell-ai>=0.0.9
      depend on sqlmodel>=0.0.21,<0.0.22, we can conclude that all of:
          ell-ai<0.0.8
          ell-ai>0.0.8
      depend on sqlmodel>=0.0.21,<0.0.22.
      And because ell-ai==0.0.8 was yanked (reason: Regression fixed in 0.0.9), we can conclude that all
      versions of ell-ai depend on sqlmodel>=0.0.21,<0.0.22.
      And because your project depends on ell-ai and sqlmodel>=0.0.22, we can conclude that your project's
      requirements are unsatisfiable.

tried to delete fastapi and used the one that comes with ell on install but even when it work to create a basic api, whein i tried to connect db is broken and not connect no matter what i tried.

ConnectionRefusedError: [Errno 111] Connection refused

uninstalling ell-ai and reinstalled fastapi resolved the connection issues. So I cannot use ell on my project unfortunately.

### PR DESCRIPTION
**Pull request by alex-dixon**
Tried making this as unobtrusive as possible :)

Studio and storage are separated and are available as optional installs.

### ell[all]
- Includes anthropic, groq, storage with sqlite, and studio (no change, same as what we have today -- this is the recommended install in the docs)
 

### ell
- Installs `ell` without studio, no tracing/storage support, and the OpenAI provider

## Complex 
Advanced users can specify extras to control what is installed and what isn't:

### ell[sqlite]
- Adds storage via sqlite
- Does not include studio

### ell[studio, postgres]
- Adds ell studio and tracing using postgres as a backend

Fixes #284

### CODE REVIEW

**Comment by MadcowD** [comment]
Would you make a PR to update the docs for this? It's great :)

File: pyproject.toml [hunk 1]
```diff
@@ -53,7 +57,18 @@ sphinx-rtd-theme = "^2.0.0"
 # causes poetry to mark it as optional = true (even if explicitly specified optional = false).
 anthropic = ["anthropic"]
 groq = ["groq"]
-all = ["anthropic", "groq"]
+sqlite = [ 'sqlmodel' ] # do we need a 'sqlite' dependency...?
```
Comment (MadcowD) [inline pyproject.toml:60]: I think no?

File: pyproject.toml [hunk 2]
```diff
@@ -53,7 +57,18 @@ sphinx-rtd-theme = "^2.0.0"
...
+postgres = ['sqlmodel', 'psychopg2']
+studio = ['fastapi', 'uvicorn', 'sqlmodel']
+all = [
+    "anthropic",
+    "groq",
+    # default storage dependencies
+    'sqlmodel',
+    # allow running stduio by default
```
Comment (MadcowD) [inline pyproject.toml:68]: Shouldn't studio be included in all?

**Comment by MadcowD** [comment]
I cannot think of a world where you would want sqlite but not studio? I think something like:
```
pip install ell-ai[all]
pip install ell-ai[studio]
pip install ell-ai[studio, postgres]
```

Would be idea.

**Review by MadcowD — CHANGES_REQUESTED** [review]
seee other comment

**Comment by alex-dixon** [comment]
@MadcowD 

> I cannot think of a world where you would want sqlite but not studio?


in pythonland it could be “my program uses a fast api version that doesn’t work with ell studio (but i still want to write traces and have versioning)”. The same goes for any dependency that is required to run studio but not write traces.

I still need to update the error messages but docs are ready. 

Any thoughts on linking to docs in the error message in addition to including an install command?

File: pyproject.toml [hunk 3]
```diff
@@ -53,7 +57,18 @@ sphinx-rtd-theme = "^2.0.0"
 # causes poetry to mark it as optional = true (even if explicitly specified optional = false).
 anthropic = ["anthropic"]
 groq = ["groq"]
-all = ["anthropic", "groq"]
+sqlite = [ 'sqlmodel' ]
+postgres = ['sqlmodel', 'psychopg2']
```
Comment (alex-dixon) [inline pyproject.toml:61]:
```suggestion
postgres = ['sqlmodel', 'psycopg2']
```

**Pull request merged**